src/pubkey.h

```diff
@@ -141,17 +141,15 @@ class CPubKey
     template <typename Stream>
     void Unserialize(Stream& s)
     {
-        unsigned int len = ::ReadCompactSize(s);
+        const unsigned int len(::ReadCompactSize(s));
         if (len <= SIZE) {
             s.read((char*)vch, len);
             if (len != size()) {
                 Invalidate();
             }
         } else {
             // invalid pubkey, skip available data
-            char dummy;
-            while (len--)
-                s.read(&dummy, 1);
+            s.ignore(len);
```

Comment:
This is the most tricky change to review I think; whether the behavior here stays the same in all cases, also in edge cases at the end of the stream.

Reply:
The only change should be in the error message and a missing call to `memcpy`.

```diff
-CDataStream::read(): end of data
+CDataStream::ignore(): end of data
```

The commit is unrelated and not needed for the changes here, so I am happy to remove. If it will be removed, I'd later on need to replace `char dummy;` with `std::byte dummy;`.

Reply:
I'm fine with keeping it. I personally think the change is correct, just mentioned it as a point of focus for reviewers.